src/header.c

```diff
@@ -6,21 +6,21 @@
 #include <endian.h>
 #include <stdio.h>
 
-
 /*!
  * \brief buildISOHeader Constructs an ISO header based on the supplied message ID and content length
+ * \param receiverID ID of the receiver of the message for which the header is to be used
  * \param id Message ID of the message for which the header is to be used
  * \param messageLength Length of the message including header and footer
  * \param debug Flag for enabling debugging
  * \return A struct containing ISO header data
  */
-HeaderType buildISOHeader(enum ISOMessageID id, uint32_t messageLength, const char debug) {
+HeaderType buildISOHeader(enum ISOMessageID id, const MessageHeaderType *input, uint32_t messageLength, const char debug) {
 	HeaderType header;
 
 	header.syncWord = ISO_SYNC_WORD;
-	header.transmitterID = getTransmitterID();
-	header.receiverID = 0; // TODO: Set receiver ID
-	header.messageCounter = 0;
+	header.transmitterID = input->transmitterID;
+	header.receiverID = input->receiverID;
+	header.messageCounter = input->messageCounter;
```

Comment:
Maybe messageCounter shall be a static variable instead that is automatically incremented each time buildISOHeader(...) is called? To make it easier for the user.

Reply:
Messagecounter must be handled by the modules using this code, as there is no way of knowing which counter is to be used